### PR TITLE
Update to latest cardano-ledger-specs.

### PR DESCRIPTION
**Pull request by nc6**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -234,8 +234,8 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/plutus
-  tag: 826c2514a40e962c2e4d56ce912803a434cc28fe
-  --sha256: 04jryrgi2wggkf4a994smbcp8j39clshbzfnvfvvk7lq0sm8kssz
+  tag: 985bd37845cd0996fc38eaccd1db1a9ad3edab3a
```
Comment (nfrisby) [inline cabal.project:237]: 3091 of `cls` uses `95e5d7488451e43062ca84d5376b3adcc465f1cd` for `plutus` -- should we match that here?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs [hunk 2]
```diff
@@ -253,7 +253,7 @@ pattern IncorrectClaimedFlag claimedFlag <-
           (Alonzo.UtxosFailure
             (Alonzo.ValidationTagMismatch
               (Alonzo.IsValid claimedFlag
-      )))))
+      ) _))))
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:256]: Minor: this feels easy to overlook to me. A couple ideas:

- rename it `_<foo>` if only so it's easier to notice

- change the layout (perhaps of the whole pattern) so the `_` is on its own line

**Review by nfrisby — COMMENTED** [review]
I wrote a couple minor comments, please consider those before merging.

Also: I just want to confirm before Approving: we're changing the serialization of all Shelley-based ledger states?

File: cabal.project (hunk 1 above)
Comment (nc6) [reply cabal.project:237]: Erm... it uses the same version: https://github.com/input-output-hk/cardano-ledger-specs/blob/309178369eebe9ce02171987f6d23c7778f7db26/cabal.project#L72

Comment (nfrisby) [reply cabal.project:237]: Aha! I glanced too quickly. `95e5d7488451e43062ca84d5376b3adcc465f1cd` is for `https://github.com/Quid2/flat.git`, not for `plutus`.

**Comment by nc6** [comment]
Yes, serialisation is changed. I think there's nothing we can do about this, we require additional information in the pulsing calculation. I'm hoping to bundle this together with other changes we make to the serialisation.

**Review by nfrisby — APPROVED** [review]
Approved. Thanks much.

**Comment by nfrisby** [comment]
I rebased onto now's `master`; it was automatic with no conflicts.

**Pull request merged**